main.tf

```diff
@@ -0,0 +1,45 @@
+/**
+ * # AWS EKS External DNS Terraform module
+ *
+ * A terraform module to deploy the [ExternalDNS](https://kubernetes-sigs.github.io/external-dns/latest) on Amazon EKS cluster.
+ *
+ * [![Terraform validate](https://github.com/lablabs/terraform-aws-eks-external-dns/actions/workflows/validate.yaml/badge.svg)](https://github.com/lablabs/terraform-aws-eks-external-dns/actions/workflows/validate.yaml)
+ * [![pre-commit](https://github.com/lablabs/terraform-aws-eks-external-dns/workflows/pre-commit.yml/badge.svg)](https://github.com/lablabs/terraform-aws-eks-external-dns/actions/workflows/pre-commit.yml)
+ *
+ */
+
+locals {
+  addon = {
+    name      = "external-dns"
+    namespace = "kube-system"
+
+    helm_chart_version = "1.15.1"
+    helm_repo_url      = "https://kubernetes-sigs.github.io/external-dns"
+  }
+
+  addon_irsa = {
+    (local.addon.name) = {
```

Comment:
Can we change the order of irsa_policy_enabled (first) and irsa_policy (second) to align with variables order within the module?